pkg/cmd/util/clientcmd/factory.go

```diff
@@ -274,14 +279,37 @@ func (c *clientCache) ClientConfigForVersion(version string) (*kclient.Config, e
 		if err != nil {
 			return nil, err
 		}
+		// TODO: We want to reuse the ustream negotiation logic, which is coupled
+		// to a concrete kube Client. The negotiation will ultimately try and
+		// build an unversioned URL using the config prefix to ask for supported
+		// server versions. If we use the default kube client config, the prefix
+		// will be /api, while we need to use the OpenShift prefix to ask for the
+		// OpenShift server versions. For now, set OpenShift defaults on the
+		// config to ensure the right prefix gets used. The client cache and
+		// negotiation logic should be refactored upstream to support downstream
+		// reuse so that we don't need to do any of this cache or negotiation
+		// duplication.
+		client.SetOpenShiftDefaults(config)
 		c.defaultConfig = config
+		defaultClient, err := kclient.New(config)
+		if err != nil {
+			return nil, err
+		}
+		c.defaultClient = defaultClient
```

Comment:
if this is only used for version negotiation, name it that, and put a TODO in the struct to remove it once we figure out negotiatio-n